Security/Http/Authenticator/OAuthAuthenticator.php

```diff
@@ -0,0 +1,250 @@
+ */
+
+namespace HWI\Bundle\OAuthBundle\Security\Http\Authenticator;
+
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwnerInterface;
+use HWI\Bundle\OAuthBundle\OAuth\State\State;
+use HWI\Bundle\OAuthBundle\Security\Core\Authentication\Token\OAuthToken;
+use HWI\Bundle\OAuthBundle\Security\Core\Exception\OAuthAwareExceptionInterface;
+use HWI\Bundle\OAuthBundle\Security\Core\User\OAuthAwareUserProviderInterface;
+use HWI\Bundle\OAuthBundle\Security\Http\ResourceOwnerMapInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\AuthenticationServiceException;
+use Symfony\Component\Security\Core\Exception\LazyResponseException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\PassportInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+use Symfony\Component\Security\Http\HttpUtils;
+
+/**
+ * OAuthAuthenticator.
+ *
+ * @author Vadim Borodavko <vadim.borodavko@gmail.com>
+ */
+class OAuthAuthenticator implements AuthenticatorInterface
+{
+    /**
+     * @var HttpUtils
+     */
+    private $httpUtils;
+
+    /**
+     * @var OAuthAwareUserProviderInterface
+     */
+    private $userProvider;
+
+    /**
+     * @var ResourceOwnerMapInterface
+     */
+    private $resourceOwnerMap;
+
+    /**
+     * @var string[]
+     */
+    private $checkPaths;
+
+    /**
+     * @var AuthenticationSuccessHandlerInterface|null
+     */
+    private $successHandler;
+
+    /**
+     * @var AuthenticationFailureHandlerInterface|null
+     */
+    private $failureHandler;
+
+    /**
+     * @var mixed[]
+     */
+    private $rawToken;
+
+    /**
+     * @var string
+     */
+    private $resourceOwnerName;
+
+    /**
+     * @var string
+     */
+    private $refreshToken;
+
+    /**
+     * @var string
+     */
+    private $createdAt;
+
+    /**
+     * Constructor.
+     *
+     * @param HttpUtils                                  $httpUtils
+     * @param OAuthAwareUserProviderInterface            $userProvider
+     * @param ResourceOwnerMapInterface                  $resourceOwnerMap
+     * @param string[]                                   $checkPaths
+     * @param AuthenticationSuccessHandlerInterface|null $successHandler
+     * @param AuthenticationFailureHandlerInterface|null $failureHandler
```

Comment:
The entire doc section can be removed because there are no real descriptions here.